### PR TITLE
Resolved sphinx build errors. Removed duplicate dense linear algegra …

### PR DESCRIPTION
**Pull request by JOCwriter**
…file. Updated toctrees.

Signed-off-by: John O'Connell <john.oconnell@intel.com>

# Description

Please include a summary of the change. Please also include relevant
motivation and context. See
[contribution guidelines](https://github.com/oneapi-src/oneMKL/blob/master/CONTRIBUTING.md)
for more details. If the change fixes an issue not documented in the project's
Github issue tracker, please document all steps necessary to reproduce it.

Fixes # (GitHub issue)

# Checklist

## All Submissions

- [ ] Do all unit tests pass locally? Attach a log.
- [ ] Have you formatted the code using clang-format?

## New interfaces

- [ ] Have you provided motivation for adding a new feature as part of RFC and
it was accepted? # (RFC)
- [ ] What version of oneAPI spec the interface is targeted?
- [ ] Complete [New features](pull_request_template.md#new-features) checklist

## New features

- [ ] Have you provided motivation for adding a new feature?
- [ ] Have you added relevant tests?

## Bug fixes

- [ ] Have you added relevant regression tests?
- [ ] Have you included information on how to reproduce the issue (either in a
      GitHub issue or in this PR)?

### CODE REVIEW

**Comment by JOCwriter** [comment]
@mkrainiuk , when I created this PR, there was no place for me to assign you as a reviewere. Can you help me get this PR properly formatted?
Thank you so much!

**Comment by mkrainiuk** [comment]
> @mkrainiuk , when I created this PR, there was no place for me to assign you as a reviewere. Can you help me get this PR properly formatted?
> Thank you so much!

Sure, I assigned it to me and Mesut as we need two reviewers.

File: docs/domains/blas/blas.rst [hunk 1]
```diff
@@ -13,5 +13,3 @@ oneMKL provides DPC++ interfaces to the Basic Linear Algebra Subprograms (BLAS)
     blas-level-3-routines.rst
     blas-like-extensions.rst
 
-
-**Parent topic:** :ref:`onemkl_dense_linear_algebra`
```
Comment (mmeterel) [inline docs/domains/blas/blas.rst:17]: Why is this removed?

Comment (JOCwriter) [reply docs/domains/blas/blas.rst:17]: Hi @mmeterel , it was removed because it was a duplicate file and not found in any toctree. I discussed with @mkrainiuk offline prior to the PR and she said it was OK to remove. The proper file is located at docs/domains/dense_linear_algebra.rst.

Docs build without errors now. Let me know if you have other concerns.

Regards,
John

Comment (mmeterel) [reply docs/domains/blas/blas.rst:17]: Hi @JOCwriter what i meant is, why the "parent topic" is removed.

Comment (JOCwriter) [reply docs/domains/blas/blas.rst:17]: Hi @mmeterel , thank you for clarifying. There is an initiative underway to align all library components to have the same look/feel and the "Parent Topic" entry is not standardized. If you look at oneDNN opensource docs for example (https://oneapi-src.github.io/oneDNN/v2/index.html), this is the direction we are moving for oneMKL. I will be making future ongoing updates to apply standard templates to all of the topics in this doc set, but this just got picked up in this PR because I was doing other work in the topic.

File: .gitignore [hunk 2]
```diff
@@ -17,3 +17,5 @@
 
 # Build
 build/
+docs/_build/
+CMakeCache.txt
```
Comment (mkrainiuk) [inline .gitignore]: I don't think we need explicitly ignore CMakeCache.txt file if you adds whole build folder to ignore. 
BTW can we align build folder with the main one and also use just "build" without underscore?

File: docs/conf.py [hunk 3]
```diff
@@ -89,7 +89,7 @@
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
-html_static_path = ['_static']
+# html_static_path = ['_static']
```
Comment (mkrainiuk) [inline docs/conf.py]: Does this line also generate warning?

Comment (JOCwriter) [reply docs/conf.py]: Hi @mkrainiuk , yes, I kept getting build warnings until I commented that line out per Todd R's recommendation.

File: .gitignore (hunk 2 above)
Comment (JOCwriter) [reply .gitignore]: @mkrainiuk , Ok, let me consult with Todd R. He just showed me how to do a sphinx build from within the docs folder and we specified the _build directory for output of docs only. If there is a way to output to the main build directory, I will be happy to do that, but let me investigate first.

File: docs/domains/blas/blas.rst (hunk 1 above)
Comment (mmeterel) [reply docs/domains/blas/blas.rst:17]: @JOCwriter thanks for clarification.

File: .gitignore (hunk 2 above)
Comment (JOCwriter) [reply .gitignore]: Hi @mkrainiuk , Ok, after talking with Todd, my next commit will address both of your concerns here.

File: docs/conf.py (hunk 3 above)
Comment (JOCwriter) [reply docs/conf.py]: Hi @mkrainiuk , my next commit will address this. I have uncommented line 92 and added a `.../docs/_static` directory with some style/formatting files in it. The docs build without warnings/errors now.

Comment (JOCwriter) [reply docs/conf.py]: @mkrainiuk , Note that with this new commit, I changed the theme in conf.py to sphinx_book_theme. I'm not sure if you have any CI dependencies where that change should be reflected (e.g. docker install manifest, etc.), so I wanted to call that out specifically.

**Comment by JOCwriter** [comment]
Hi @mkrainiuk, please let me know if you have any further changes. I think this PR is good to merge upon your approval.
Regards,
John

**Review by mkrainiuk — APPROVED** [review]

**Review by mmeterel — APPROVED** [review]
Overall, this looks good to me. 

However, this repo is missing LAPACK and RNG documentation and needs to be added.

**Comment by mkrainiuk** [comment]
> Overall, this looks good to me.
> 
> However, this repo is missing LAPACK and RNG documentation and needs to be added.

Agree, we need to add them.
